sr_health_check/src/sr_health_check/sr_health_check_gui.py

```diff
@@ -75,37 +85,70 @@ def __init__(self, context):
+        self._setup_connections()
         self._timer.start(100)
 
         self.display_data()
+        self.set_side_selection_accesibility()
 
         self._rqt_node_name = rospy.get_name()
-        rospy.Subscriber("/rosout", Log, self.status_subscriber)
-
-        self._bold_font = QFont()
-        self._bold_font.setBold(True)
-
-    def status_subscriber(self, msg):
+        self._log_subcriber = rospy.Subscriber("/rosout", Log, self._status_subscriber)
+        self._log_publisher = rospy.Publisher("/rosout", Log, queue_size=1)
+
+    def set_side_selection_accesibility(self):
+        '''
+            Sets the visibility of side selection radio buttons.
+        '''
+        checkable_side_selection_radio_buttons = []
+        joint_states_msg = rospy.wait_for_message("/joint_states", JointState, timeout=2)
+        joint_names = [joint_name.split("_")[0] for joint_name in joint_states_msg.name]
+
+        for side_prefix in SrHealthCheck._SIDE_PREFIXES:
+            if side_prefix in joint_names:
+                checkable_side_selection_radio_buttons.append(side_prefix)
+                self._detected_prefixes.append(side_prefix)
+
+        if 'lh' in checkable_side_selection_radio_buttons:
+            self._widget.side_left_radio_button.setChecked(True)
+            self._widget.side_left_radio_button.click()
+        if 'rh' in checkable_side_selection_radio_buttons:
+            self._widget.side_right_radio_button.setChecked(True)
+            self._widget.side_right_radio_button.click()
+
+    def _status_subscriber(self, msg):
+        '''
+            Callback method for /rosout subscriber.
+            @param msg: rosgraph_msgs.msg.Log type message
+        '''
         if msg.name == self._rqt_node_name:
             status = msg.msg
             self._widget.label_status.setText(f"Status:{status}")
 
     def timerEvent(self):  # pylint: disable=C0103
-        checks_are_running = False
-        for check_name in self._check_names:
-            if self._checks_to_execute[check_name]['thread'].is_alive():
-                checks_are_running = True
+        '''
+            Timer event to enable/disable the start checks butons.
+        '''
         selected_checks = any(check for check in self._check_names if self._selected_checks[check])
-        self._widget.button_start_selected.setEnabled(not checks_are_running and selected_checks)
-        self._widget.button_start_all.setEnabled(not checks_are_running)
+        self._widget.button_start_selected.setEnabled(not self._checks_running and selected_checks)
+        self._widget.button_start_all.setEnabled(not self._checks_running)
+        self._widget.button_stop.setEnabled(self._checks_running)
+        self._widget.side_right_radio_button.setEnabled(not self._checks_running and "rh" in self._detected_prefixes)
+        self._widget.side_left_radio_button.setEnabled(not self._checks_running and "lh" in self._detected_prefixes)
+
+    def are_checks_running(self):
```

Comment:
Shouldn't this function read the status of the queue as the documentation says? If not, perhaps the documentation should be changed.

Reply:
Removed this method as I am directly checking the status through a variable in the code